### PR TITLE
Ports the Portable Seed Extractor from TG and adds it to the H.U.E.Y. upgrade

### PR DESCRIPTION
**Pull request by jknpj**
https://streamable.com/7whwr

:cl:
 * rscadd: Ports the portable seed extractor from TG. It doubles ad both plant bag and seed extractor but it can only extract at most two seeds per crop.
 * rscdel: Removed the plant bag from the H.U.E.Y. upgrade.
 * rscadd: Added the portable seed extractor to the H.U.E.Y. upgrade.

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1694,3 +1694,52 @@ Game Mode config tags:
 	if(!istype(T2))
 		T2 = get_turf(B)
 	return sqrt(((T2.x - T1.x) ** 2) + ((T2.y - T1.y) ** 2))
+
+/proc/seedify(obj/item/O, obj/machinery/seed_extractor/extractor, mob/living/user)
+	if(!O)
+		return FALSE
```
Comment (PJB3005) [inline code/__HELPERS/unsorted.dm]: Should do a `CRASH()` instead because this is clearly an error.

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -1694,3 +1694,52 @@ Game Mode config tags:
 	if(!istype(T2))
 		T2 = get_turf(B)
 	return sqrt(((T2.x - T1.x) ** 2) + ((T2.y - T1.y) ** 2))
+
+/proc/seedify(obj/item/O, obj/machinery/seed_extractor/extractor, mob/living/user)
```
Comment (PJB3005) [inline code/__HELPERS/unsorted.dm]: `extractor` and `user` should default to null if they're optional.

File: code/game/objects/items/weapons/storage/bags.dm [hunk 3]
```diff
@@ -125,6 +125,34 @@ obj/item/weapon/storage/bag/plasticbag/quick_store(var/obj/item/I)
...
+		return
+
+	if(P.contents)
+		var/played = FALSE
+		for(var/obj/item/I in P.contents)
+			if(seedify(I) && !played)
+				playsound(get_turf(P), 'sound/machines/juicerfast.ogg', 50, 1)
+				played = TRUE
+		P.orient2hud(usr)
+		if(usr)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/storage/bags.dm]: `usr` aint happening.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/storage/bags.dm (hunk 3 above)
Comment (jknpj) [reply code/game/objects/items/weapons/storage/bags.dm]: fug

File: code/game/objects/items/weapons/storage/bags.dm [hunk 4]
```diff
@@ -137,8 +137,14 @@ obj/item/weapon/storage/bag/plasticbag/quick_store(var/obj/item/I)
 
 /datum/action/item_action/dissolve_contents/Trigger()
 	var/obj/item/weapon/storage/bag/plants/portactor/P = target
+	var/mob/user = usr
```
Comment (PJB3005) [inline code/game/objects/items/weapons/storage/bags.dm]: Wait. You mean action buttons actually don't pass a user?

TG why.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**